contracts/escrow-fee-distributor/tests/integration.rs

```diff
@@ -0,0 +1,703 @@
+        &user1,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // sets 200 * 1000_000 xASTRO tokens to user2
+    BaseAstroportTestPackage::mint(
+        router_ref,
+        base_pack.staking.clone().unwrap().address,
+        xastro_token.clone(),
+        &user2,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // checks if user2's xASTRO token balance is equal to 200 * 1000_000
+    BaseAstroportTestPackage::check_balance(
+        router_ref,
+        &xastro_token.clone(),
+        &user2,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // locks 100 vxASTRO from user1 for WEEK * 2
+    base_pack
+        .create_lock(router_ref, user1.clone(), WEEK * 2, 100)
+        .unwrap();
+
+    // locks 200 vxASTRO from user2 for WEEK * 2
+    base_pack
+        .create_lock(router_ref, user2.clone(), WEEK * 2, 200)
+        .unwrap();
+
+    // try set checkpoint from user1 when it is disabled
+    let err = router_ref
+        .execute_contract(
+            user1.clone(),
+            base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &ExecuteMsg::CheckpointToken {},
+            &[],
+        )
+        .unwrap_err();
+
+    assert_eq!("Checkpoint token is not available!", err.to_string());
+
+    // try set checkpoint from owner
+    router_ref
+        .execute_contract(
+            owner.clone(),
+            base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &ExecuteMsg::CheckpointToken {},
+            &[],
+        )
+        .unwrap();
+
+    // check if tokens per week is set
+    let resp: Vec<Uint128> = router_ref
+        .wrap()
+        .query_wasm_smart(
+            &base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &QueryMsg::FeeTokensPerWeek {},
+        )
+        .unwrap();
+    assert_eq!(vec![Uint128::new(100)], resp);
+
+    // going to the next week
+    router_ref.update_block(next_block);
+    router_ref.update_block(|b| b.time = b.time.plus_seconds(WEEK));
+
+    // sets 900 ASTRO tokens to distributor (simulate receive astro from maker)
+    BaseAstroportTestPackage::mint(
+        router_ref,
+        owner.clone(),
+        base_pack.astro_token.clone().unwrap().address,
+        &base_pack.escrow_fee_distributor.clone().unwrap().address,
+        900,
+    );
+
+    // try to claim some fee when is checkpoint per week is disabled
+    router_ref
+        .execute_contract(
+            user1.clone(),
+            base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &ExecuteMsg::Claim { recipient: None },
+            &[],
+        )
+        .unwrap();
+
+    // check if voting supply per week is set
+    let resp: Vec<Uint128> = router_ref
+        .wrap()
+        .query_wasm_smart(
+            &base_pack.escrow_fee_distributor.clone().unwrap().address,
+            &QueryMsg::VotingSupplyPerWeek {},
+        )
+        .unwrap();
+    assert_eq!(vec![Uint128::new(1442307), Uint128::new(721153),], resp);
```

Comment:
Voting power supply for the first week should be equal to 100 * (1 + 1.5 * 2 / 104) + 200 * (1 + 1.5 * 2 / 104) = 308.653846. Note that voting power has the same presicion as xASTRO has. For example, 1.1 xASTRO = 1100000. You need to divide voting power by 10^token.decimals